### DISCUSSION TITLE
Advice on migrating single server with docker containers to Proxmox

### DISCUSSION THREAD

**Discussion by proddy** [tteck/Proxmox#804] [Help Needed]
Hi, this repo looks exactly what I need so kudos @tteck for all the effort you've put in. Donation on its way!

I'm looking for some advice. Currently, I have a NUC8 with 32GB RAM and 1TB SSD, running Ubuntu 22.10 a hacked version of Home Assistant Supervised and everything else (plex, mariadb, mosquitto mqtt, radarr/sonarr, portainter etc..) as docker containers. About 15 of these are in a docker compose file. It's been running fine for the last few years but as Home Assistant is becoming a more closed system, it's getting increasingly hard to keep it supported with all the workarounds. So I was planning to migrate to Debian and start over and then found your posting on https://community.home-assistant.io/t/installing-home-assistant-os-using-proxmox-7

What would the best way to set up my system with Proxmox, using your scripts?
A. Run HA as Home Assistant OS VM, and also plex, mariadb, mqtt etc as individual LXCs also using your scripts?
B. Run HA as Home Assistant OS VM, and continue to run all my other services under a single Docker LXC

thanks

**Comment by tteck** [comment]
> What would the best way to set up my system with Proxmox

Well, that's the good thing about Home Assistant, the ***Best*** varies depending on the user. Proxmox makes it easy to run HA one way while experimenting with other methods at the same time to determine which you like better. I'd say the most popular install is Home Assistant OS, an all in one solution. But it sounds like you may need more (may not) than a simple HAOS install. 

I personally run Home Assistant Core and all of the other services in separate LXC containers. I have no Docker containers running.

![Screenshot 2022-12-04 6 22 22 AM](https://user-images.githubusercontent.com/21342670/205487814-ded65d51-87a1-4081-a2fc-c4e27b9bdc5f.png)
 
I prefer the detached approach.  Nothing depends on HA to be running. Most of my automation's can still run without HA. Z2m > MQTT > NR. Another reason is, it's easier to backup each service individually just in case I goof something up, a simple restore from a backup/snapshot and I'm back in business. Troubleshooting is much easier also. But that's my use case, it may not be best for you. Id suggest starting out with what you're most comfortable with, and go from there.

**Reply by tteck** [reply]
No matter which way you go, backups, backups and backups. Setup a good backup policy. I backup every 4 hours to an external USB NVME drive formatted fat32 (so any PC can access the files) with a retention of 24 (4 days)

![Screenshot 2022-12-04 7 18 21 AM](https://user-images.githubusercontent.com/21342670/205490246-d313b6f8-edba-47e3-9665-6c8d01cc5723.png)

It has saved my butt many times.

**Comment by proddy** [comment]
Thanks for the explanation. I agree, using Proxmox is a good example to try out all the flavours of HA. The HA supervisor was handy for backups and keeping HA up-to-date but that can be easily be done outside too so I'll start with 'core'. Also keeping everything separate and independent is a great approach as you mentioned. I've only ever used docker for this. And great idea to keep backups on an external USB drive too, copying that too!

So, I'd better get cracking. I'll report back here on how it ends up, thanks again!

**Reply by tteck** [reply]
It may be better to start with HAOS beings that you have backups (hopefully) to restore and get back up running quick. 
It can be rough listening to all of the nagging from the family because things are not working.
Then start experimenting with core.

**Reply by proddy** [reply]
My HA is actually quite bare except for a few automation scripts which I have in yaml files. And I do like building things clean from scratch. I think the only thing I need to keep is the historic data from the Energy dashboard which takes gas/electricity consumption for my P1 meter via esphome and this is all stored in a separate MariaDB.

**Reply by proddy** [reply]
I moved everything over to proxmox using your scripts and having a blast. I'll report back findings here. 

One thing is that I had to install the python3 mysqlclient in the homeassistant-core LXC to be able to use MariaDB as HA's recorder. But it could have been an err on my side.

```
apt install libmariadb-dev-compat
pip3 install mysqlclient
```

**Comment by dmshimself** [comment]
I started with HA as a VM using the scripts here (and they are wonderful), but I found after a while that on my low spec server, quite a few resources were being used and reliability wasn't what I wanted.  But being able to do an easy backup on one rig and restore to another using HA was very handy.  

So my first step was to leave the HASSOS VM pretty much as it was and use the mariabdb, zwave, mqtt and esphome LXC scripts and shift the configs across one at a time.  This quite dramatically affected the resources, especially CPU, of the HA rig.

I then left things for a while and created a new HA LXC, , added an extra DB to the mariabdb, copied the config files across from the HASSOS instance, and renamed the automation.yaml so it would not be used and gradually brought both systems up to much the same config.  I wasn't in any rush and some of my HASSOS config was pretty out of date.  The device/entities were sometimes different names etc.

Then when I was happy, over a weekend I renamed the automations.yaml on the LXC and  edited to use the right devices/entities that had changed.  Inevitably there were a few things I had forgotten, but it didn't take as long as I thought and things have been running very well for some time.  Having both systems running at the same time for a while helped a lot.

The individual LXCs are very easy to backup and in the case of mariadb so is the data.

One day backups from the HA/settings of the HA docker will be re-storable, but you wouldn't be going along this route if you didn't know how to work with the backups you can already do :-)

If I had started with a powerful server and I might have left my HASSOS running, but things seems more stable now and restart times are significantly quicker because the LXC containers for the DB etc are just left running.
